pyro/nn/auto_reg_nn.py

```diff
@@ -110,8 +108,12 @@ class AutoRegressiveNN(nn.Module):
 
     :param input_dim: the dimensionality of the input
     :type input_dim: int
-    :param hidden_dim: the dimensionality of the hidden units
-    :type hidden_dim: int
+    :param hidden_dims: the dimensionality of the hidden units per layer
+    :type hidden_dims: list[int]
+    :param count_params: number of parameters to return from a forward call of the network.
+    :type count_params: int
+    :param param_dim: the dimensionality of each parameter
```

Comment:
param_dim: the dimensionality of each parameter

should this be a list for more flexibility?

also this param description could be a bbit more informative?

Reply:
e.g. is this clear from the docstring?

# Shape the output into self.count_params of dimension self.param_dim*self.input_dim

Reply:
I've put something along the lines of what you suggest